### PR TITLE
fix: render the `uid` for all messages in a channel

### LINKED ISSUE #141: A message's `uid` is not rendered when a subscribe/publish block has a single message associated with it
#### Describe the bug
When a channel's subscribe/publish block has a single message associated with it the `uid` is not being rendered. It doesn't look like this matches the behaviour of the `html-template` (see below). This behaviour is not ideal for our current use case but the motivation for this issue is what I think is an inconsistency between renderers. The rendering of `uid` is not necessarily the best property to output but the `title` property is also not output anywhere.

#### How to Reproduce
There is an explicit check to the parser [here](https://github.com/asyncapi/markdown-template/blob/master/components/Channels.js#L74) to check for multiple messages and the `uid` is not rendered if there is not multiple.

#### Expected behaviour
Having had a look at the html-template and the underlying react components (some of which the `markdown-template` replaces) I would expect the `uid` to be rendered whether there are single or multiple possible messages.

I'm happy to put together a PR if the expected behaviour is desirable.

### PR DESCRIPTION
**Pull request by stuartforrest-infinity**
**Description**
Using the `markdown-template` renderer when a channel's subscribe/publish block has a single message associated with it the `uid` was not being rendered. This did not match the behaviour of the `html-tmplate` renderer. This PR resolves this inconsistency and will always render a message's `uid` regardless of how many messages there are.

**Related issue(s)**
Resolves #141

### CODE REVIEW

**Comment by stuartforrest-infinity** [comment]
This is a tiny change and I've eyeballed the output is as expected when run against [this](https://github.com/asyncapi/generator/blob/master/test/docs/dummy.yml). The file being tested against as part of the `test:generation` npm job is an older version of this one, I'm not fully clear on the testing approach to the renders beyond the eyeballing the output but this may want updating?

File: components/Channels.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ function Operation({ operation }) {
           ))}
         </>
       ) : (
-        <Message title='Message' message={operation.message(0)} />
+        <Message title={`Message \`${operation.message(0)}\``} message={operation.message(0)} />
```
Comment (magicmatatjahu) [inline components/Channels.js:84]: Currently you render whole object and you end with output like:

```md
##### Message `[object Object]`

*This is just a dummy info message*

...
```

you should pass to the title `prop` only `uid` value like:

```suggestion
        <Message title={`Message \`${operation.message(0).uid()}\``} message={operation.message(0)} />
```

and then you will have:

```md
##### Message `dummyInfo`

...
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
Thanks for contribution @stuartforrest-infinity! Please refer to mu suggestion. When it comes to testing, don't worry, leave it as it is.

**Comment by magicmatatjahu** [comment]
@stuartforrest-infinity Ok, thanks for propagating my changes, however we have one last problem. Sometimes `message` can be empty, so we have to change the code for single message to the:

```js
      ) : operation.message(0) ? (
        <Message title={`Message \`${operation.message(0).uid()}\``} message={operation.message(0)} />
      ) : null}
```

to handle also case with empty message. Could you propagate it in your PR?

**Comment by stuartforrest-infinity** [comment]
That is a little inconvenient. Your suggested change would change the output of the rendered because the current behaviour is even if there isn't a message a title should be rendered - https://github.com/asyncapi/markdown-template/blob/master/components/Message.js#L10.

To maintain the current behaviour I've pushed a change to conditionally render the `uid` instead if you can take a look.

I'm not sure the current behaviour makes sense but changing it is out of scope of this issue.

**Comment by magicmatatjahu** [comment]
@stuartforrest-infinity If message doesn't exist, so for what render standalone `Message` title? For that I suggest to render message only when message object exists, so:

```js
      ) : operation.message(0) ? (
        <Message title={`Message \`${operation.message(0).uid()}\``} message={operation.message(0)} />
      ) : null}
```

is good.

**Comment by stuartforrest-infinity** [comment]
Done :)

**Comment by stuartforrest-infinity** [comment]
I'll do some refactoring to get rid of the nested ternary a bit later :)

**Comment by magicmatatjahu** [comment]
@stuartforrest-infinity Yeah sure :) You can create separate component like below to avoid complex ternary operators:

```jsx
function Operation({ operation }) {
  const type = operation.isPublish() ? 'publish' : 'subscribe';
  return (
    <Text>
      <Header type={4}>{`\`${type}\` Operation`}</Header>
      {operation.summary() && (
        <Text newLines={2}>
          *{operation.summary()}*
        </Text>
      )}
      {operation.hasDescription() && (
        <Text newLines={2}>
          {operation.description()}
        </Text>
      )}
      <OperationMessage operation={operation} />
    </Text>
  );
}

function OperationMessage({ operation }) {
  if (operation.hasMultipleMessages()) {
    return (
      <>
        <Text newLines={2}>
          Accepts **one of** the following messages:
        </Text>
        {operation.messages().map(msg => (
          <Message title={`Message \`${msg.uid()}\``} message={msg} />
        ))}
      </>
    );
  }

  if (operation.message(0)) {
    return (
      <Message title={`Message \`${operation.message(0).uid()}\``} message={operation.message(0)} />
    );
  }

  return null;
}
```

Thanks!

**Review by magicmatatjahu — APPROVED** [review]
@stuartforrest-infinity Thanks for contribution! :)

**Comment by magicmatatjahu** [comment]
@all-contributors please add @stuartforrest-infinity for bug, code

**Pull request merged**